From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rishwanth Yeddula <rish@cpanel.net>
Date: Thu, 12 Jul 2018 13:53:29 -0500
Subject: [PATCH 2/2] Rebuild configure with the additional LDFLAG for Brotli,
 H2, and SSL

---
 configure | 61 ++++++++++++++++++++++++++++++++-----------------------
 1 file changed, 36 insertions(+), 25 deletions(-)

diff --git a/configure b/configure
index 74711b4..637b4b2 100755
--- a/configure
+++ b/configure
@@ -980,6 +980,9 @@ CURL_LT_SHLIB_USE_NO_UNDEFINED_TRUE
 CURL_LT_SHLIB_USE_VERSION_INFO_FALSE
 CURL_LT_SHLIB_USE_VERSION_INFO_TRUE
 LT_SYS_LIBRARY_PATH
+LD_BROTLI
+LD_H2
+SSL_LDFLAGS
 OTOOL64
 OTOOL
 LIPO
@@ -1098,7 +1101,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -1215,6 +1217,9 @@ LDFLAGS
 LIBS
 CPPFLAGS
 CPP
+SSL_LDFLAGS
+LD_H2
+LD_BROTLI
 LT_SYS_LIBRARY_PATH'
 ac_subdirs_all='ares'
 
@@ -1254,7 +1259,6 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
@@ -1507,15 +1511,6 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -1653,7 +1648,7 @@ fi
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
 		datadir sysconfdir sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-		libdir localedir mandir runstatedir
+		libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -1806,7 +1801,6 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -2056,6 +2050,9 @@ Some influential environment variables:
   CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
               you have headers in a nonstandard directory <include dir>
   CPP         C preprocessor
+  SSL_LDFLAGS User-defined LDFLAGS for SSL.
+  LD_H2       User-defined LDFLAGS for nghttp2.
+  LD_BROTLI   User-defined LDFLAGS for brotli.
   LT_SYS_LIBRARY_PATH
               User-defined run-time library search path.
 
@@ -7264,7 +7261,7 @@ else
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7309,7 +7306,7 @@ else
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7332,7 +7329,7 @@ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7376,7 +7373,7 @@ else
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7399,7 +7396,7 @@ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -14030,6 +14027,12 @@ need_version=unknown
 
 
 
+
+
+
+
+
+
 case $host_os in
 aix3*)
   version_type=linux # correct to gnu/linux during the next big refactor
@@ -21038,7 +21041,9 @@ $as_echo "found" >&6; }
 
     if test -n "$PREFIX_BROTLI"; then
     LIB_BROTLI="-lbrotlidec"
-    LD_BROTLI=-L${PREFIX_BROTLI}/lib$libsuff
+    if test -z $LD_BROTLI; then
+        LD_BROTLI=-L${PREFIX_BROTLI}/lib$libsuff
+    fi
     CPP_BROTLI=-I${PREFIX_BROTLI}/include
     DIR_BROTLI=${PREFIX_BROTLI}/lib$libsuff
   fi
@@ -22335,8 +22340,10 @@ $as_echo "$as_me: PKG_CONFIG_LIBDIR will be set to \"$OPENSSL_PCDIR\"" >&6;}
 
             LIB_OPENSSL="$PREFIX_OPENSSL/lib$libsuff"
     if test "$PREFIX_OPENSSL" != "/usr" ; then
-      SSL_LDFLAGS="-L$LIB_OPENSSL"
-      SSL_CPPFLAGS="-I$PREFIX_OPENSSL/include"
+      if test -z $SSL_LDFLAGS; then
+          SSL_LDFLAGS="-L$LIB_OPENSSL"
+          SSL_CPPFLAGS="-I$PREFIX_OPENSSL/include"
+      fi
     fi
     SSL_CPPFLAGS="$SSL_CPPFLAGS -I$PREFIX_OPENSSL/include/openssl"
     ;;
@@ -22479,12 +22486,14 @@ $as_echo "found" >&6; }
     fi
          $PKGCONFIG --libs-only-l openssl 2>/dev/null`
 
-      SSL_LDFLAGS=`
+      if test -z $SSL_LDFLAGS; then
+        SSL_LDFLAGS=`
     if test -n "$OPENSSL_PCDIR"; then
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
          $PKGCONFIG --libs-only-L openssl 2>/dev/null`
+      fi
 
       SSL_CPPFLAGS=`
     if test -n "$OPENSSL_PCDIR"; then
@@ -27621,15 +27630,17 @@ $as_echo "$as_me: -l is $LIB_H2" >&6;}
     { $as_echo "$as_me:${as_lineno-$LINENO}: -I is $CPP_H2" >&5
 $as_echo "$as_me: -I is $CPP_H2" >&6;}
 
-    LD_H2=`
+    if test -z $LD_H2; then
+        LD_H2=`
     if test -n "$want_h2_path"; then
       PKG_CONFIG_LIBDIR="$want_h2_path"
       export PKG_CONFIG_LIBDIR
     fi
 
-      $PKGCONFIG --libs-only-L libnghttp2`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: -L is $LD_H2" >&5
+          $PKGCONFIG --libs-only-L libnghttp2`
+        { $as_echo "$as_me:${as_lineno-$LINENO}: -L is $LD_H2" >&5
 $as_echo "$as_me: -L is $LD_H2" >&6;}
+    fi
 
     LDFLAGS="$LDFLAGS $LD_H2"
     CPPFLAGS="$CPPFLAGS $CPP_H2"
